### ISSUE TITLE
StringFile call in testinstall is broken on cygwin

### ISSUE THREAD

**Issue by alex-konovalov** [gap-system/gap#881]
This originates from #875 (which is based on #862). I have checked that the suggestion by @frankluebeck to fix compiler warnings works (at least on 32-bit Cygwin - don't know what happens on Cygwin64 yet). However, then I discovered while testing that GAP does not load because of an error in `StringFile`:
```
$ bin/gap.sh -r -A tst/testinstall.g
 ┌───────┐   GAP 4.9.0, 07-Aug-2016, build of 2016-08-07 15:42:48 (GMTDT)
 │  GAP  │   http://www.gap-system.org
 └───────┘   Architecture: i686-pc-cygwin-gcc-default32
 Libs used:  gmp
 Loading the library and packages ...
Error, in StringFile: no error
 at /proc/cygdrive/C/gap4r9/lib/string.gi:758 called from
<function "StringFile">( <arguments> )
 called from read-eval loop at /proc/cygdrive/C/gap4r9/pkg/GAPDoc-1.5.1/lib/GAPDoc2LaTeX.gi:210
you can 'quit;' to quit to outer loop, or
you can 'return;' to continue
brk>
```
On Windows, this happens both when GAP is started in Cygwin shell, and on a Cygwin-free machine.

**Comment by fingolfin** [comment]
@alex-konovalov Try commenting out the two `SySetErrorNo();` calls, recompile and rerun the test -- does it now give a more meaningful error? Perhaps we should also print the error code, not just the message.

**Comment by alex-konovalov** [comment]
@fingolfin thanks. This is now fixed by @markuspf in #884.
